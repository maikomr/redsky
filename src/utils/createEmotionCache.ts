import createCache from "@emotion/cache";

/*
 Ensures that Emotionâ€™s default settings will be replaced with our custom styles 
 and that this information will be configured both on the client and server sides.
*/
export default function createEmotionCache() {
  return createCache({
    key: "css",
    prepend: true, // this makes our custom styles load first
  });
}
