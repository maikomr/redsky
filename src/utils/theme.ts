import { createTheme } from "@mui/material/styles";

export const theme = createTheme({
  palette: {
    primary: { main: "#52796f" },
    secondary: { main: "#84a98c" },
  },
});
