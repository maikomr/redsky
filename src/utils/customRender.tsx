import { ThemeProvider } from "@mui/material/styles";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render, RenderOptions } from "@testing-library/react";
import React, { ReactElement } from "react";
import { theme } from "./theme";

const queryClient = new QueryClient();

interface AllTheProvidersProps {
  children: ReactElement;
}

function AllTheProviders({ children }: AllTheProvidersProps): JSX.Element {
  return (
    <ThemeProvider theme={theme}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  );
}

export const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });
