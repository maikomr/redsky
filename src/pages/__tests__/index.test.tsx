import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "@/pages/index";
import { customRender } from "@/utils/customRender";

jest.mock("next/router", () => require("next-router-mock"));

describe("Home", () => {
  it("renders unchanged", () => {
    const { container } = customRender(<Home />);
    expect(container).toMatchSnapshot();
  });
});
