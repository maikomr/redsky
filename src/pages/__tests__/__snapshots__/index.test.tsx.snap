// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders unchanged 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="css-678bp8-MuiStack-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-bz1l3-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            placeholder="Search location here..."
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1qtk8s1-MuiSkeleton-root"
      />
      <div
        class="css-1kwgaa7-MuiStack-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1u5pjy3-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1u5pjy3-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1u5pjy3-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1u5pjy3-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1u5pjy3-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1u5pjy3-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-1u5pjy3-MuiSkeleton-root"
        />
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
      >
        Copyright © 2023
      </p>
    </div>
  </div>
</div>
`;
